include/zephyr/drivers/auxdisplay.h

```diff
@@ -0,0 +1,824 @@
+	auxdisplay_cursor_position_set_api cursor_position_set;
+	auxdisplay_cursor_position_get_api cursor_position_get;
+	auxdisplay_display_position_set_api display_position_set;
+	auxdisplay_display_position_get_api display_position_get;
+	auxdisplay_capabilities_get_api capabilities_get;
+	auxdisplay_clear_api clear;
+	auxdisplay_brightness_get_api brightness_get;
+	auxdisplay_brightness_set_api brightness_set;
+	auxdisplay_backlight_get_api backlight_get;
+	auxdisplay_backlight_set_api backlight_set;
+	auxdisplay_is_busy_api is_busy;
+	auxdisplay_custom_character_set_api custom_character_set;
+	auxdisplay_write_api write;
+	auxdisplay_custom_command_api custom_command;
+};
+
+/**
+ * @brief		Set cursor enabled status on a text display
+ *
+ * @param dev		Pointer to device structure
+ * @param enabled	True to enable cursor, false to disable
+ *
+ * @retval		0 on success else negative errno code.
+ */
+__syscall int auxdisplay_cursor_set_enabled(const struct device *dev,
+					    bool enabled);
+
+static inline int z_impl_auxdisplay_cursor_set_enabled(const struct device *dev,
+						       bool enabled)
+{
+	struct auxdisplay_driver_api *api = (struct auxdisplay_driver_api *)dev->api;
+
+	if (!api->cursor_set_enabled) {
+		return -ENOSYS;
+	}
+
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	pm_device_state_get(dev, &state);
+
+	if (state == PM_DEVICE_STATE_SUSPENDED) {
+		return -EIO;
+	}
+#endif
+
+	return api->cursor_set_enabled(dev, enabled);
+}
+
+/**
+ * @brief		Set cursor blinking status on a text display
+ *
+ * @param dev		Pointer to device structure
+ * @param enabled	Set to true to enable blinking position, false to disable
+ *
+ * @retval		0 on success else negative errno code.
+ */
+__syscall int auxdisplay_position_blinking_set_enabled(const struct device *dev,
+						       bool enabled);
+
+static inline int z_impl_auxdisplay_position_blinking_set_enabled(const struct device *dev,
+								  bool enabled)
+{
+	struct auxdisplay_driver_api *api = (struct auxdisplay_driver_api *)dev->api;
+
+	if (!api->position_blinking_set_enabled) {
+		return -ENOSYS;
+	}
+
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	pm_device_state_get(dev, &state);
+
+	if (state == PM_DEVICE_STATE_SUSPENDED) {
+		return -EIO;
+	}
+#endif
+
+	return api->position_blinking_set_enabled(dev, enabled);
+}
+
+/**
+ * @brief		Set cursor shift after character write and display shift
+ *
+ * @param dev		Pointer to device structure
+ * @param direction	Sets the direction of the display when characters are written
+ * @param display_shift	If true, will shift the display when characters are written
+ *			(which makes it look like the display is moving, not the cursor)
+ *
+ * @retval		0 on success else negative errno code.
+ */
+__syscall int auxdisplay_cursor_shift_set(const struct device *dev,
+					  uint8_t direction, bool display_shift);
+
+static inline int z_impl_auxdisplay_cursor_shift_set(const struct device *dev,
```

Comment:
I think we should follow the same structure everywhere: `set_enabled` vs. `shift_set`. @carlescufi do we have guidelines on this?

Reply:
It does follow it really, `auxdisplay_position_blinking_set_enabled_api` = `position blinking` is being `set` if it is `enabled` or not -> `auxdisplay_cursor_shift_set_api` = `cursor shift` is being `set` for the `direction` and `shift` (direction and display_shift would equate to enabled)